win32ss/user/ntuser/ntstubs.c

```diff
@@ -647,14 +647,51 @@ NtUserQueryInformationThread(IN HANDLE ThreadHandle,
     return Status;
 }
 
-DWORD
+DWORD_PTR
 APIENTRY
 NtUserQueryInputContext(
     HIMC hIMC,
-    DWORD dwUnknown2)
+    DWORD dwType)
 {
-    TRACE("NtUserQueryInputContext(%p, 0x%lX)\n", hIMC, dwUnknown2);
-    return 0;
+    PIMC pIMC;
+    PTHREADINFO ptiIMC;
+    DWORD_PTR ret = 0;
+
+    UserEnterExclusive();
+
+    if (!IS_IMM_MODE())
+        goto Quit;
+
+    pIMC = UserGetObject(gHandleTable, hIMC, TYPE_INPUTCONTEXT);
+    if (!pIMC)
+        goto Quit;
+
+    ptiIMC = pIMC->head.pti;
+
+    switch (dwType)
+    {
+        case 0:
```

Comment:
You should give names to these types.
Something like, for example (I don't know what's the best naming convention we can use, with respect to the other flags we have):
`QIC_PROCESSID`